patient-dao/src/main/java/hu/kuncystem/patient/dao/appointment/JDBCAppointmentDao.java

```diff
@@ -15,31 +34,261 @@
+    // place.(just, the sql condition is different)
+    private static final String SQL_SELECT_APPOINTMENT = "SELECT " + "at.id," + "at.appointment," + "at.description,"
+            + "at.sid," + "STRING_AGG(an.note, ',') AS notes," + "u1.id AS doctor_id,"
+            + "u1.user_name AS doctor_user_name," + "u1.passw AS doctor_passw," + "u1.fullname AS doctor_fullname,"
+            + "u1.email AS doctor_email," + "u1.active AS doctor_active," + "u2.id AS patient_id,"
+            + "u2.user_name AS patient_user_name," + "u2.passw AS patient_passw," + "u2.fullname AS patient_fullname,"
+            + "u2.email AS patient_email," + "u2.active AS patient_active " + "FROM appointment_table at "
+            + "INNER JOIN users u1 ON (u1.id = at.doctor_id) " + "INNER JOIN users u2 ON (u2.id = at.patient_id) "
+            + "LEFT JOIN appointment_notes an ON (an.appointment_id = at.id) " + " WHERE $1$ "
+            + "GROUP BY at.id, u1.id, u2.id;";
+
+    private static final String SQL_DELETE_APPOINTMENT = "DELETE FROM appointment_table WHERE id = ?;";
+
+    private static final String SQL_DELETE_APPOINTMENT_NOTES = "DELETE FROM appointment_notes WHERE appointment_id = ?;";
+
+    private static final String SQL_UPDATE = "UPDATE appointment_table "
+            + "SET doctor_id = ?, patient_id = ?, appointment = ?, description = ? WHERE id = ?;";
+
+    @Autowired
+    private JdbcOperations jdbc;
+
+    public boolean deleteAppointment(Appointment appointment) throws DatabaseException {
+        try {
+            // delete appointment notes if these are exists
+            jdbc.update(SQL_DELETE_APPOINTMENT_NOTES, appointment.getId());
+        } catch (DataAccessException e) {
+            throw new DatabaseException(
+                    DatabaseException.STRING_DATA_ACCESS_EXCEPTION + " " + SQL_DELETE_APPOINTMENT_NOTES, e);
+        }
+        try {
+            // delete appointment
+            if (jdbc.update(SQL_DELETE_APPOINTMENT, appointment.getId()) == 0) {
+                // not found the appointment by id
+                return false;
+            } else {
+                return true;
+            }
+        } catch (DataAccessException e) {
+            throw new DatabaseException(DatabaseException.STRING_DATA_ACCESS_EXCEPTION + " " + SQL_DELETE_APPOINTMENT,
+                    e);
+        }
+
     }
 
-    public List<Appointment> getAppointments(User user, Date dateFrom, Date dateTo) {
-        // TODO Auto-generated method stub
-        return null;
+    public Appointment getAppointment(long id) throws DatabaseException {
+        // replace the joker characters to concrete sql
+        String sql = SQL_SELECT_APPOINTMENT.replace("$1$", "at.id = ?");
+        Appointment appointment = null;
+        try {
+            appointment = jdbc.queryForObject(sql, new AppointmentRowMapper(), id);
+        } catch (EmptyResultDataAccessException e) {
+            appointment = new Appointment();

```

Comment:
I believe I was not clear enough the last time; it's OK to return null when querying a single object; we should avoid that when returning a Collection (Set/List).